### PR TITLE
Feature/incorporate genre data

### PR DESCRIPTION
**Pull request by m1073496**
## What does this PR do (summary of changes)?
 - Makes a network request to fetch genres data
 - Adds genre data to corresponding song object to display on UI
 - Refactors to pull network requests into helper file and DRY up code
 
## How should it be tested?
 - UI should display song details including genres (make sure the genres look right for some songs)
 
## Future Iterations:
 - Possible refactor of modifyData function in App component

### CODE REVIEW

**Comment by m1073496** [comment]
[Hook up API calls to app](https://trello.com/c/nYRubsoZ/72-hook-up-api-calls-to-app)

File: src/App.js [hunk 1]
```diff
@@ -21,32 +21,52 @@ class App extends Component {
   }
 
   componentDidMount() {
-    fetchAllSongs()
-      .then(data => Promise.all(data))
+    fetchAllSongData()
       .then(data => {
-        this.setState({ songBook: data })
-        this.setState({renderedSongs: [...this.state.songBook]})
+        this.modifyData(data[0], data[1])
```
Comment (elizhahn) [inline src/App.js]: What if we destructured the data here:

 const [songs, genres] = data
        this.modifyData(songs, genres)

Just for readability purposes

**Review by elizhahn — APPROVED** [review]

File: src/App.js (hunk 1 above)
Comment (m1073496) [reply src/App.js]: you got it!

File: src/App.js [hunk 2]
```diff
@@ -21,32 +21,48 @@ class App extends Component {
...
+
+        accu[i].genres = foundGenres.map(genre => genre[0]);
+        return accu;
+      }, []);
+
+      this.setState({ songBook: modifiedData });
+      this.setState({ renderedSongs: [...this.state.songBook] })
+  }
+
+
```
Comment (elizhahn) [inline src/App.js:47]: nice refactor taking out the promise.all here

**Review by elizhahn — APPROVED** [review]

**Pull request merged**